core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2867,33 +2883,43 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
 
   void execute(bool arg) {
     destroy = arg;
+
+    // Shortcut for zero initialization
+    ValueType value{};
+    if (std::is_trivial<ValueType>::value &&
+        std::is_trivially_copy_assignable<ValueType>::value &&
+        Impl::is_zero_byte(value) && !destroy) {
+      (void)ZeroMemset<ExecSpace, ValueType*, typename DeviceType::memory_space,
+                       Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+          space,
+          Kokkos::View<ValueType*, typename DeviceType::memory_space,
+                       Kokkos::MemoryTraits<Kokkos::Unmanaged>>(ptr, n),
+          value);
+      return;
+    }
+
     PolicyType policy(0, n);
     std::string functor_name;
-    if (!space.in_parallel()) {
```

Comment:
This is for use cases involving Views of Views, we need that branch, I believe

Reply:
But then it should still be a `KOKKOS_FUNCTION`, doesn't it? Anyway, I will just separate it into a different pull request.

Reply:
Sure. For reference, it isn't KOKKOS_FUNCTION, because the _outer_ constructor still happens on the Host. It's really confusing, everything about View of Views is weird